src/libs/runtime-utils.js

```diff
@@ -164,10 +169,24 @@ export const machineCost = machineType => {
 
 export const currentApp = _.flow(trimAppsOldestFirst, _.last)
 
-export const currentDataDisk = (app, galaxyDataDisks) => {
+export const currentAttachedDataDisk = (app, galaxyDataDisks) => {
   return _.find({ name: app?.diskName }, galaxyDataDisks)
 }
 
+export const getCurrentPersistentDiskName = (apps, galaxyDataDisks) => {
+  const currentGalaxyApp = currentApp(apps)
+  const currentDataDiskName = currentGalaxyApp?.diskName
+  const attachedDataDiskNames = _.without([undefined], _.map(app => app.diskName, apps))
+  return currentDataDiskName ?
+    _.find({ currentDataDiskName }, galaxyDataDisks) :
+    _.last(_.sortBy('auditInfo.createdDate', _.filter(({ name, status }) => status !== 'Deleting' && !_.includes(name, attachedDataDiskNames), galaxyDataDisks)))
```

Comment:
I'm having to spend a lot of time to wrap my head around the logic here. Need to think more about whether we could make it easier to parse.

Reply:
yeah it took me a bit to understand as well - we do the same kind of logic for determining whether or not to delete a PD for a GCE VM. see https://github.com/DataBiosphere/terra-ui/blob/dev/src/components/NewRuntimeModal.js#L169